server/src/main/java/org/elasticsearch/action/admin/indices/shrink/TransportResizeAction.java

```diff
@@ -88,32 +88,43 @@ protected void masterOperation(Task task, final ResizeRequest resizeRequest, fin
         // there is no need to fetch docs stats for split but we keep it simple and do it anyway for simplicity of the code
         final String sourceIndex = indexNameExpressionResolver.resolveDateMathExpression(resizeRequest.getSourceIndex());
         final String targetIndex = indexNameExpressionResolver.resolveDateMathExpression(resizeRequest.getTargetIndexRequest().index());
+
+        final IndexMetadata sourceMetadata = state.metadata().index(sourceIndex);
+        if (sourceMetadata == null) {
+            listener.onFailure(new IndexNotFoundException(sourceIndex));
+            return;
+        }
+
         IndicesStatsRequestBuilder statsRequestBuilder = client.admin().indices().prepareStats(sourceIndex).clear().setDocs(true);
         IndicesStatsRequest statsRequest = statsRequestBuilder.request();
         statsRequest.setParentTask(clusterService.localNode().getId(), task.getId());
+        // TODO: only fetch indices stats for shrink type resize requests and move request validation (outside of doc counts) to before
```

Comment:
Failing the listener on a missing index isn't strictly necessary here because I added the `try-catch` around `prepareCreateIndexRequest` anyway but I figured it more nicely motivates this `TODO` and follow-up cleanup to the logic here that previously relied on the exceptions just bubbling up all the way.

Reply:
Fair point this is slightly broken to begin with I guess. I'll drop that part of the TODO, I still think it'd be valid to avoid fetching the shard stats needlessly.